homeassistant/components/bluetooth/util.py

```diff
@@ -15,6 +16,38 @@
     WINDOWS_DEFAULT_BLUETOOTH_ADAPTER,
     AdapterDetails,
 )
+from .models import BluetoothServiceInfoBleak
+
+
+async def async_load_history_from_system() -> dict[str, BluetoothServiceInfoBleak]:
+    """Load the device and advertisement_data history if available on the current system."""
+    if platform.system() != "Linux":
+        return {}
+    from bluetooth_adapters import (  # pylint: disable=import-outside-toplevel
+        BlueZDBusObjects,
+    )
+
+    bluez_dbus = BlueZDBusObjects()
+    await bluez_dbus.load()
+    now = time.monotonic()
+    return {
+        address: BluetoothServiceInfoBleak(
+            name=history.advertisement_data.local_name
+            or history.device.name
```

Comment:
I remember someone told me once that we shouldn't have multi-line inline expressions? 🤷🏻

Reply:
I generally try to avoid the multi line ternary expression but I think the ors are fine as you don't have to backtrack to read them.